### DISCUSSION TITLE
Debug shell profile

### DISCUSSION THREAD

**Discussion by electriquo** [XQuartz/XQuartz#376] [Q&A]
After sifting through the issues, I found out that my issue was caused by something in my shell profile (e.g. `~/.bash_profile`). When the profile is causing the issue, XQuartz says nothing about it in its logs (`~/Library/Logs/X11/org.xquartz.log`) and I couldn't find what in my profile make XQuartz not to start.

Do you have any recommendation on how to debug the profile issue? Such as finding which line causes the issue? (for instance, see that output of `bash -x` when the profile is loaded by XQuartz?, which bash binary does XQuartz uses? etc.)

**Comment by dazuelos** [comment]
In a standard `Terminal` enter:
```
echo ${PATH}
```
and add it to your original question.

Most probably you changed your `PATH`

`XQuartz` is using the shell determined by the symbolic link in:
```
/private/var/select
```

To further debug your starting of XQuartz, I suggest you the following way:
in a standard `Terminal` enter:

```
/bin/sh -x /opt/X11/bin/startx -- /opt/X11/bin/Xquartz
```
